beacon-chain/sync/subscriber.go

```diff
@@ -158,6 +158,10 @@ func (r *Service) subscribe(topic string, validator pubsub.Validator, handle sub
 				continue
 			}
 
+			if msg.ReceivedFrom == r.p2p.PeerID() {
```

Comment:
Would this be a good regression test? If yes, I'll be more than happy to open a good first issue

@prestonvanloon